compiler/resolution/functionResolution.cpp

```diff
@@ -7272,9 +7309,108 @@ static void resolveAutoCopies() {
       resolveAutoCopy(ts->type);
       resolveAutoDestroy(ts->type);
     }
+
+    if (isRecord(ts->type) ) {
+      // Mark record/tuple types as POD or NOT_POD
+      propagateNotPOD(ts->type);
+    }
   }
 }
 
+/* In order to correctly initialize records or tuples in which
+   a component has FLAG_IGNORE_NOINIT, we need to propagate that
+   flag to the parent types as well. While doing so, we also
+   propagate whether or not that type is not Plain-old-Data
+   (which would mean it does not need auto copies or auto
+    destroys - bit copies will do).
+
+   After this function is called on a record type, that type and any field
+   record types must have either FLAG_POD or FLAG_NOT_POD set.
+
+   After setting either FLAG_POD or FLAG_NOT_POD if necessary,
+   returns true if FLAG_NOT_POD is set, false otherwise.
+
+   This function should only be called during resolution.
+   Call isPOD (or check FLAG_POD/FLAG_NOT_POD) after resolution.
+ */
```

Comment:
I wonder whether we should stick with the convention of having only // comments.